src/peakrdl_sv/regmodel.py

```diff
@@ -22,8 +52,357 @@ def _parse_rdl(self, rdlfile: str) -> AddressMap:
+            )
+
+        return {
+            reg_name: {
+                field.inst_name: FieldWrapper(field, field.reset or 0) for field in reg
+            }
+            for reg_name, reg in self._reg_map.items()
+        }
+
+    # --------------------------------------------------------------------------------
+    # Utilities
+    # --------------------------------------------------------------------------------
+
+    def get_register_by_name(self, reg_name: str) -> Register | None:
+        """Gets a register by its relative path name, '.' separated per tier of the
+        hierarchy
+
+        :param reg_name: The register name as a . separated string
+        :type reg_name: str
+        :return: The target register, None if the register name is not found in the
+        regmap
+        :rtype: Register | None
+        """
+        try:
+            return self._reg_map[reg_name]
+        except KeyError:
+            self._log.info(f"Could not find register in {self._reg_map.keys()}")
+            return None
+
+    def split_value_over_fields(
+        self,
+        target: Register,
+        value: int,
+    ) -> Generator[tuple[Any, int], Any, None]:
+        """Split a value over the fields of a register with correct masking depending
+        on the field location
+
+        :param target: The register to target
+        :type target: Register
+        :param value: The value to write to the registers
+        :type value: int
+        :yield: tuples of the field address and the value to write
+        :rtype: Generator[tuple[Any, int], Any, None]
+        """
+
+        # format the write value an `regwidth`-bit binary string
+        bits = f"{value:0{target.regwidth}b}"
+
+        # Extract the bits we are going to write and conver them to int
+        for field in target:
+            # get the slice indices of the field
+            lower = target.regwidth - field.msb - 1
+            upper = target.regwidth - field.lsb
+
+            # perform the slice (python slicing non inclusive)
+            yield (field.absolute_address, int(bits[lower:upper], 2))  # noqa: E203
+
+    # --------------------------------------------------------------------------------
+    # Generic read and write wrappers
+    # --------------------------------------------------------------------------------
+
+    async def write(
+        self,
+        reg_name: str,
+        field_name: str | None = None,
+        data: int | None = None,
+        **kwargs,
+    ) -> None:
+        """Writes the value of the DUT register. Two options for calling:
+
+        (1) register name + data -> write to named register a particular value
+        (2) register name -> write desired value to a named register
+
+        :param reg_name: Register name to write to
+        :type reg_name: str | int
+        :param data: Data to write to register, defaults to None
+        :type data: int | None, optional
+        """
+
+        if isinstance(field_name, int):
```

Comment:
This should/could be some form of an `ArgumentError`.  Perhaps a `TypeError` is the correct way to do it?

Reply:
Yeah, i'll raise `TypeError`. This is to catch the fact that you might be tempted to write
```
RAL.write('reg_name', 0x1234)
```
but because we accept a field name as the second argument, the correct convention is actually
```
RAL.write('reg_name', data=0x1234)
```
when you want to write an entire register.